prov/gni/include/gnix_cq.h

```diff
@@ -45,6 +45,9 @@
 #define GNIX_CQ_DEFAULT_FORMAT struct fi_cq_entry
 #define GNIX_CQ_DEFAULT_SIZE   256
 
+/* forward declaration */
+struct gnix_fid_ep;
```

Comment:
I tried that but the compiler then warns: `/prov/gni/include/gnix_cq.h:86:59: warning: 'struct gnix_fid_ep' declared inside parameter list will not be visible outside of this definition or declaration
 ssize_t _gnix_cq_add_event(struct gnix_fid_cq *cq, struct gnix_fid_ep *ep,...`